plugin/src/gatsby-node.ts

```diff
@@ -1,257 +1,203 @@
-import type { GatsbyNode } from 'gatsby';
-import type { Callable } from '@cometjs/core';
-import type { Source } from '@graphql-tools/utils';
-import type { GatsbyStore } from './gatsby-utils';
-
-import path from 'path';
-import { stripIndent } from 'common-tags';
-import { parseGraphQLSDL } from '@graphql-tools/utils';
-import { gqlPluckFromCodeString } from '@graphql-tools/graphql-tag-pluck';
-
-import {
-  writeFile,
-  readFile,
-  deduplicateFragmentFromDocuments,
-} from './common';
-import {
-  setupCodegenWorker,
-  setupEmitSchemaWorker,
-  setupInsertTypeWorker,
-} from './workers';
-import {
-  requirePluginOptions,
-  RequiredPluginOptions,
-  GRAPHQL_TAG_PLUCK_OPTIONS,
-} from './plugin-utils';
-
-// Plugin will track documents what is actually used by Gatsby.
-const trackedSource = new Map<string, Source>();
+import type { GatsbyNode, Reporter } from 'gatsby';
+import { interpret } from 'xstate';
+import { Kind } from 'gatsby/graphql';
+
+import type { TypegenReporter } from './internal/reporter';
+import { validateConfig } from './internal/config';
+import { typegenMachine } from './internal/machine';
+import { makeEmitSchemaService } from './services/emitSchema';
+import { makeEmitPluginDocumentService } from './services/emitPluginDocument';
+import { makeAutofixService } from './services/autofix';
+import { makeCodegenService } from './services/codegen';
+import type { PluginOptions } from './types';
+import { readFileContent, writeFileContent } from './utils';
+
+export const pluginOptionsSchema: GatsbyNode['pluginOptionsSchema'] = ({
+  Joi,
+}) => {
+  const documentOutputOptionsSchema = Joi.object({
+    format: Joi.string()
+      .valid('introspection', 'sdl')
+      .default('sdl')
+      .required(),
+    commentDescriptions: Joi.boolean()
+      .default(true),
+  }).required();
+
+  return Joi.object({
+    language: Joi.string()
+      .valid('typescript', 'flow')
+      .default('typescript'),
+    namespace: Joi.string()
+      .default('GatsbyTypes'),
+    outputPath: Joi.string()
+      .default('src/__generated__/gatsby-types.d.ts'),
+    includeResolvers: Joi.boolean()
+      .default(false),
+    autoFix: Joi.boolean()
```

Comment:
You'll want to remove this old key, Gatsby will automatically warn about invalid keys

When I tried out the RC I got `warn [typegen] The `autoFix` option is deprecated. Use `autofix` instead.` so I think it gets set due to this here